src/snowflake/snowpark/session.py

```diff
@@ -440,7 +443,13 @@ def __init__(
         self._session_stage = random_name_for_temp_object(TempObjectType.STAGE)
         self._stage_created = False
 
-        if isinstance(conn, MockServerConnection):
+        # Two possibilities here:
+        # 1. create another MockUDFRegistrationSandbox class
+        # 2. weave the sandbox code from within the original UDFRegistration class
+        if (
+            isinstance(conn, MockServerConnection)
+            and not _is_execution_environment_sandboxed
```

Comment:
Seems incorrect. If the sandboxing is enabled as well as local testing is enabled on the current session, then we won't create a mock registration instance.